pyanaconda/storage/checker.py

```diff
@@ -411,6 +412,51 @@ def verify_mounted_partitions(storage, constraints, report_error, report_warning
                                "installation. Please unmount it and retry.") % part.path)
 
 
+def verify_lvm_destruction(storage, constraints, report_error, report_warning):
+    """ Verify that destruction of LVM devices is correct.
+
+    This means: If a PV is destroyed, the VG on it must be destroyed too.
+    Additionally, destroying an incomplete VG warns about this.
+
+    :param storage: a storage to check
+    :param constraints: a dictionary of constraints
+    :param report_error: a function for error reporting
+    :param report_warning: a function for warning reporting
+    """
+    # Implementation of the verification uses an inverted approach: Check that all PVs to be
+    # destroyed are contained in a VG that will be destroyed too. If there is a PV to be destroyed
+    # that has a VG that isn't going to be destroyed, that is an error. This is because we ask
+    # blivet to ignore some disks, and these might contain a PV. Crucially, blivet implements this
+    # such that the ignoring hides also all dependent devices - for any disk, it could be a PV
+    # partition, which would then hide also all VGs and LVs that depend on the PV. It does not
+    # matter that the VG needs other PVs too - the hiding wins.
+    log.debug("called verify_lvm_destruction()")
+    destroyed_vg_names = set()
+    vg_names_seen_in_destroyed_pvs = set()
+    for action in storage.devicetree.actions:
+        if isinstance(action, ActionDestroyDevice) and action.device.type == "lvmvg":
```

Comment:
You can use `action.is_destroy` here.

Reply:
Ah, thanks for the tip - actually I think `action.is_destroy and action.is_device` would be better?

Reply:
Yes, that works, and is_format for the other one. Changed.